docs/user_guide/datasets.rst

```diff
@@ -0,0 +1,394 @@
+It's featured in `scipy lectures <https://scipy-lectures.org/packages/scikit-learn/auto_examples/plot_boston_prediction.html>`_, 
+indexed in the `University of California-Irvine Machine Learning Repository 
+<https://archive.ics.uci.edu/ml/machine-learning-databases/housing/>`_ and in 
+Carnegie Mellon University's `StatLib <http://lib.stat.cmu.edu/datasets/boston>`_, 
+and for a time was included as one of scikit-learn's
+ and tensorflow's standard toy datasets. 
+ 
+It has also been the benchmark of choice for `many <https://ieeexplore.ieee.org/abstract/document/8556738/>`_ 
+`machine <https://lib.dr.iastate.edu/cgi/viewcontent.cgi?article=1187&context=imse_conf>`_ 
+`learning <https://proceedings.neurips.cc/paper/1999/file/f3144cefe89a60d6a1afaf7859c5076b-Paper.pdf>`_
+`papers <https://arxiv.org/search/?query=boston+housing&searchtype=all&source=header>`.
+
+In scikit-learn version 1.2, the dataset will be removed.
+
+The dataset contains the following columns:
+
+============ ==========================================================================
+Column Name   Description                                                              
+============ ==========================================================================
+CRIM         per capita crime rate by town                                         
+ZN           proportion of residential land zoned for lots over 25,000 sq.ft.
+INDUS        proportion of non-retail business acres per town
+CHAS         Charles River dummy variable (= 1 if tract bounds river; 0 otherwise)
+NOX          nitric oxides concentration (parts per 10 million)
+RM           average number of rooms per dwelling
+AGE          proportion of owner-occupied units built prior to 1940
+DIS          weighted distances to five Boston employment centers
+RAD          index of accessibility to radial highways
+TAX          full-value property-tax rate per $10,000
+PTRATIO      pupil-teacher ratio by town
+B            1000(Bk - 0.63)^2 where Bk is the proportion of blacks by town
+LSTAT        % lower status of the population
+MEDV         Median value of owner-occupied homes in $1000’s
+============ ============================================================================
+
+The cells below show basic summary statistics about the data, the data types of the 
+columns, and the number of missing values.
+
+Note that the :func:`fairlearn.datasets.fetch_boston` function warns user by 
+default that the dataset contains fairness issues. 
+
+Setting :code:`warn=False` will turn the warning off. 
+
+To return the dataset as a :class:`pandas.DataFrame`, pass 
+:code:`as_frame=True` and call the *data* attribute.
+
+
+For more information about how to use the :code:`fetch_boston` function, 
+visit :mod:`fairlearn.datasets`. 
+
+.. note::
+
+    Calling the :func:`fairlearn.datasets.fetch_boston` function will raise a 
+    :code:`FairnessWarning`.
+    For more information on this warning refer to :mod:`fairlearn.datasets`.
```

Comment:
This is a bit too close to the API reference for my taste. I'd just keep the part about "for more information about how to use the function visit <link to API reference>.

Reply:
I have removed the note.